src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/RazorLoggerFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CodeAnalysis.Razor.Logging;
+
+[Export(typeof(IRazorLoggerFactory)), System.Composition.Shared]
+internal class RazorLoggerFactory : IRazorLoggerFactory
+{
+    private readonly ILoggerFactory _loggerFactory;
+
+    [ImportingConstructor]
+    public RazorLoggerFactory([ImportMany(typeof(IRazorLoggerProvider))] IEnumerable<IRazorLoggerProvider> razorLoggerProviders)
+    {
+        _loggerFactory = LoggerFactory.Create(b =>
+        {
+            // We let everything through, and expect individual loggers to control their own levels
+            b.AddFilter(level => true);
```

Comment:
Is this needed?

Reply:
Yes, otherwise the deafult will prevent debug logs from appearing, even if the user picks that from the options

Reply:
weird